### ISSUE TITLE
intermittent 404 on https://mines.legumeinfo.org/legumemine/service/web-properties?format=json brings down the server

### ISSUE THREAD

**Issue by adf-ncgr** [legumeinfo/graphql-server#102]
per discussion started [over here ](https://github.com/soybase/jekyll-soybase/pull/52#issuecomment-1813335638)
I believe I was seeing something similar (but more frequent) when trying to test on a non-production VM, which I had chalked up to network flakiness on that VM. I'm not clear on why "the web-properties URL is pinged constantly" but clearly the reported error (copied again below) is coming from tomcat so it seems like it's not a networking issue. 

```
graphql-server-110-graphql-server-1  | /app/node_modules/@apollo/datasource-rest/dist/RESTDataSource.js:95
graphql-server-110-graphql-server-1  |         return new graphql_1.GraphQLError(`${response.status}: ${response.statusText}`, {
graphql-server-110-graphql-server-1  |                ^
graphql-server-110-graphql-server-1  |
graphql-server-110-graphql-server-1  | GraphQLError: 404: 404
graphql-server-110-graphql-server-1  |     at IntermineAPI.errorFromResponse (/app/node_modules/@apollo/datasource-rest/dist/RESTDataSource.js:95:16)
graphql-server-110-graphql-server-1  |     at IntermineAPI.throwIfResponseIsError (/app/node_modules/@apollo/datasource-rest/dist/RESTDataSource.js:87:26)
graphql-server-110-graphql-server-1  |     at /app/node_modules/@apollo/datasource-rest/dist/RESTDataSource.js:199:32
graphql-server-110-graphql-server-1  |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
graphql-server-110-graphql-server-1  |     at async IntermineAPI.fetch (/app/node_modules/@apollo/datasource-rest/dist/RESTDataSource.js:231:56)
graphql-server-110-graphql-server-1  |     at async IntermineAPI.get (/app/node_modules/@apollo/datasource-rest/dist/RESTDataSource.js:111:17)
graphql-server-110-graphql-server-1  |     at async IntermineAPI.webProperties (file:///app/dist/data-sources/intermine/intermine.server.js:54:16)
graphql-server-110-graphql-server-1  |     at async IntermineAPI.verifyIntermineVersion (file:///app/dist/data-sources/intermine/api/index.js:265:32) {
graphql-server-110-graphql-server-1  |   path: undefined,
graphql-server-110-graphql-server-1  |   locations: undefined,
graphql-server-110-graphql-server-1  |   extensions: {
graphql-server-110-graphql-server-1  |     response: {
graphql-server-110-graphql-server-1  |       url: 'https://mines.legumeinfo.org/legumemine/service/web-properties?format=json',
graphql-server-110-graphql-server-1  |       status: 404,
graphql-server-110-graphql-server-1  |       statusText: '404',
graphql-server-110-graphql-server-1  |       body: '<!doctype html><html lang="en"><head><title>HTTP Status 404 – Not Found</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 404 – Not Found</h1><hr class="line" /><p><b>Type</b> Status Report</p><p><b>Message</b> The requested resource [&#47;legumemine&#47;service&#47;web-properties] is not available</p><p><b>Description</b> The origin server did not find a current representation for the target resource or is not willing to disclose that one exists.</p><hr class="line" /><h3>Apache Tomcat/9.0.80</h3></body></html>'
graphql-server-110-graphql-server-1  |     }
graphql-server-110-graphql-server-1  |   }
graphql-server-110-graphql-server-1  | }
graphql-server-110-graphql-server-1  |
graphql-server-110-graphql-server-1  | Node.js v20.9.0
```

**Comment by adf-ncgr** [comment]
this is a bandaid, but is there any reason not to add a restart: always directive to the compose.prod.yml for this?

**Comment by alancleary** [comment]
> this is a bandaid, but is there any reason not to add a restart: always directive to the compose.prod.yml for this?

I wouldn't put it into the repo but it seems a reasonable solution for now.

**Comment by adf-ncgr** [comment]
OK, not sure why it wouldn't be appropriate for the version in the repos, but have just done it locally for now.

**Comment by sammyjava** [comment]
Yeah, the graphql-server needs to be resilient to mine outages, obviously. Sometimes LegumeMine drops out (research ongoing) but also I sometimes do a restart on the mines if I've updated the data in some way. (The constant polling of web-properties seems a bit over the top to me, it's certainly annoying when you run graphql-server in dev mode and get log lines output like crazy to the console.)

**Comment by alancleary** [comment]
> OK, not sure why it wouldn't be appropriate for the version in the repos, but have just done it locally for now.

You caught me on my way out the door yesterday so when you said "bandaid" my mind immediately went to the proverb, "nothing is more permanent than a temporary solution." I gave you my gut reaction and was on my way.

Adding restart always to `compose.prod.yml` seems reasonable. Let's keep it out of `compose[.dev].yml`, though.

> Yeah, the graphql-server needs to be resilient to mine outages, obviously.

Agreed. Just because one data source has gone down doesn't mean the whole server should come down. Having the production container always restart is a good fail-safe. I'll open an issue for this.

> The constant polling of web-properties seems a bit over the top to me, it's certainly annoying when you run graphql-server in dev mode and get log lines output like crazy to the console.

Yeah, it's super annoying. When I implemented the Intermine version check (which is the source of the polling) I tried to avoid this but the way Apollo Server works made it hard to avoid without some code mangling. I just told it to check the version when an Intermine data source is instantiated; the polling is a side-effect of how Apollo manages its data sources. It's probably worth revisiting at some point...

**Comment by alancleary** [comment]
@adf-ncgr Mind opening a PR that makes the restart always change to the `compose.dev.yml` file?

**Comment by adf-ncgr** [comment]
sure, but I thought we had agreed it would be only to the compose.prod.yml file- right?

**Comment by alancleary** [comment]
> sure, but I thought we had agreed it would be only to the compose.prod.yml file- right?

Yes. Thanks for confirming!

**Comment by alancleary** [comment]
Temporary fix implemented in PR #111. Follow issue #103 for a more robust fix.
